src/main/kotlin/com/terning/server/kotlin/application/internshipAnnouncement/InternshipAnnouncementService.kt

```diff
@@ -0,0 +1,61 @@
+package com.terning.server.kotlin.application.internshipAnnouncement
+
+import com.querydsl.core.Tuple
+import com.terning.server.kotlin.application.internshipAnnouncement.dto.HomeAnnouncementsResponse
+import com.terning.server.kotlin.domain.filter.Filter
+import com.terning.server.kotlin.domain.filter.FilterRepository
+import com.terning.server.kotlin.domain.internshipAnnouncement.InternshipAnnouncementRepository
+import com.terning.server.kotlin.domain.internshipAnnouncement.exception.InternshipAnnouncementErrorCode
+import com.terning.server.kotlin.domain.internshipAnnouncement.exception.InternshipAnnouncementException
+import com.terning.server.kotlin.domain.user.User
+import com.terning.server.kotlin.domain.user.UserRepository
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.Pageable
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+@Transactional(readOnly = true)
+class InternshipAnnouncementService(
+    private val internshipRepository: InternshipAnnouncementRepository,
+    private val userRepository: UserRepository,
+    private val filterRepository: FilterRepository,
+) {
+    fun getFilteredAnnouncements(
+        userId: Long,
+        sortBy: String,
+        pageable: Pageable,
+    ): HomeAnnouncementsResponse {
+        val user =
+            userRepository.findById(userId)
+                .orElseThrow {
+                    InternshipAnnouncementException(InternshipAnnouncementErrorCode.NOT_FOUND_USER_EXCEPTION)
+                }
+
+        val filter = filterRepository.findLatestByUser(user)
+        val pagedTuples = fetchAnnouncements(user, filter, sortBy, pageable)
+
+        if (pagedTuples.isEmpty) {
+            return HomeAnnouncementsResponse(
+                totalPages = 0,
+                totalCount = 0,
+                hasNext = false,
+                announcements = emptyList(),
+            )
+        }
+
+        return HomeAnnouncementsResponse.from(pagedTuples)
+    }
+
+    private fun fetchAnnouncements(
+        user: User,
+        filter: Filter?,
+        sortBy: String,
+        pageable: Pageable,
+    ): Page<Tuple> {
+        if (filter == null || filter.isDefault()) {
+            return internshipRepository.findAllInternshipsWithScrapInfo(user, sortBy, pageable)
+        }
+        return internshipRepository.findFilteredInternshipsWithScrapInfo(user, filter, sortBy, pageable)
+    }
```

Comment:
return문을 조금 더 코틀린스럽게 해볼 순 없을까 고민하다가 아래처럼 해도 동일한 리턴값을 가질 수 있을 것 같아서 한번 제안해봐요! 
```suggestion
    ): Page<Tuple> =
        if (filter == null || filter.isDefault()) {
            internshipRepository.findAllInternshipsWithScrapInfo(
                user = user,
                sortBy = sortBy,
                pageable = pageable,
            )
        } else {
            internshipRepository.findFilteredInternshipsWithScrapInfo(
                user = user,
                filter = filter,
                sortBy = sortBy,
                pageable = pageable,
            )
        }
```

Reply:
와, 유빈님! 이렇게 더 좋은 코드를 위해 함께 고민해주시고 제안까지 남겨주셔서 정말 감동이에요! 🥹

제안해주신 것처럼 `if-else` 표현식을 사용하면 함수를 하나의 표현식으로 만들 수 있어서 정말 코틀린스럽고 간결해지는 것 같아요. 유빈님 덕분에 저도 코드를 보는 새로운 관점을 또 하나 배웠어요. 정말 감사합니다!

제가 `else`를 사용하지 않고 `if`에서 `return`을 먼저 하도록 코드를 작성했던 건 나름의 이유가 있었답니다! 😊

저는 '조건이 맞으면 여기서 함수를 끝낸다'는 `가드 클로즈(Guard Clause)` 패턴을 적용하고 싶었어요. `else` 없이 간단한 케이스나 예외 케이스를 먼저 처리하고 빠져나가면, 그 아래의 코드는 자연스럽게 주된 로직에만 집중할 수 있게 되거든요. 나중에 코드를 다시 읽을 때 `else` 블록까지 따라가지 않아도 되어서 흐름을 이해하기 조금 더 편할 거라고 생각했어요.

물론 이건 코드를 짜는 사람의 스타일에 따른 차이라서 유빈님이 제안해주신 방법도 정말 훌륭한 코드예요! 👍

괜찮으시다면, 이번에는 코드의 흐름을 명확하게 하고 싶었던 제 의도를 담아서 현재 코드를 유지하는 건 어떨까요? 유빈님의 좋은 제안은 앞으로 다른 함수에 코드를 작성할 때 꼭 참고해서 적용해보고 싶어요!

함께 더 좋은 코드를 만들어나갈 수 있도록 깊이 고민해주셔서 다시 한번 정말 감사합니다!